starter-code/src/App.js

```diff
@@ -1,18 +1,113 @@
-import React, { Component } from 'react';
-import logo from './logo.svg';
-import './App.css';
+import React, { Component } from "react";
+import logo, { ReactComponent } from "./logo.svg";
+import "./App.css";
+import contacts from "./contacts.json";
+const firstFive = [];
+for (let i = 0; i < 5; i++) {
+  firstFive.push(contacts[i]);
+}
+//console.log(contacts.length);
+
+//console.log(randomNumber);
+class Table extends Component {
+  state = {
+    firstContacts: firstFive
+  };
+  add = () => {
+    let randomNumber = Math.floor(Math.random() * 199 + 5);
```

Comment:
instead of 199, use contacts.length to make it dynamic, just keep in mind that you have to dubtract 5 afterwards to not go out of the array:

Math.floor(Math.random() * contacts.length + 5)-5

Reply:
Also keep in mind that you never check if the contact is already in your this.state.firstContacts meaning you may get an error with your key={elem.id} as the id may thus be similar for some contacts